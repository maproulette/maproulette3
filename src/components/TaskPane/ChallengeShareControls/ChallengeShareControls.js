import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { ShareButtons, generateShareIcon } from 'react-share'
import './ChallengeShareControls.scss'

export default class ChallengeShareControls extends Component {
  render() {
    if (!this.props.challenge) {
      return null
    }

    const {
      FacebookShareButton,
      TwitterShareButton,
      EmailShareButton,
    } = ShareButtons

    const FacebookIcon = generateShareIcon('facebook')
    const TwitterIcon = generateShareIcon('twitter')
    const EmailIcon = generateShareIcon('email')

    const shareUrl = `${process.env.REACT_APP_URL}/browse/challenges/${this.props.challenge.id}`
    const title = process.env.REACT_APP_TITLE
    const hashtag = 'maproulette'

    return (
      <div className={classNames('challenge-share-controls', this.props.className)}>
        <div className="share-control">
          <FacebookShareButton url={shareUrl} quote={title}>
            <span className="share-icon">
              <FacebookIcon size={26} round />
            </span>
          </FacebookShareButton>
        </div>

        <div className="share-control">
          <TwitterShareButton url={shareUrl} hashtags={[hashtag]} title={title}>
            <span className="share-icon">
              <TwitterIcon size={26} round />
            </span>
          </TwitterShareButton>
        </div>

        <div className="share-control">
          <EmailShareButton url={shareUrl} subject={title} body={shareUrl}>
            <span className="share-icon">
              <EmailIcon size={26} round />
            </span>
          </EmailShareButton>
        </div>
      </div>
    )
  }
}

ChallengeShareControls.propTypes = {
  challenge: PropTypes.object,
}
