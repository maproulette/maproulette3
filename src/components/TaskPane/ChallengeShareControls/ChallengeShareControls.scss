.challenge-share-controls {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  flex-wrap: wrap;
  overflow: hidden;

  .share-control {
    margin-right: 15px;
    line-height: 0;

    &:hover {
      cursor: pointer;
    }

    .SocialMediaShareButton:focus {
      outline: 0;
    }
  }

  &.is-minimized {
    flex-direction: column;

    .share-control {
      margin-right: 0;
      margin-bottom: 15px;
    }
  }
}
