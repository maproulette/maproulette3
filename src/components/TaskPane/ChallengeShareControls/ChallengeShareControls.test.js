import React from 'react'
import ChallengeShareControls from './ChallengeShareControls'
import _cloneDeep from 'lodash/cloneDeep'

const propsFixture = {
  challenge: {
    id: 123,
  },
}

let basicProps = null

beforeEach(() => {
  basicProps = _cloneDeep(propsFixture)
})

test("renders with props as expected", () => {
  const wrapper = shallow(
    <ChallengeShareControls {...basicProps} />
  )

  expect(wrapper.find('.challenge-share-controls').exists()).toBe(true)
  expect(wrapper.find('.share-icon').length).toBe(3)
  expect(wrapper).toMatchSnapshot()
})

test("does not explode with null challenge", () => {
  const wrapper = shallow(
    <ChallengeShareControls />
  )

  expect(wrapper).toMatchSnapshot()
})

test("renders with props className in encapsulating div", () => {
  const wrapper = shallow(
    <ChallengeShareControls {...basicProps} className="my-test"/>
  )

  expect(wrapper.find('.my-test').exists()).toBe(true)
  expect(wrapper).toMatchSnapshot()
})
