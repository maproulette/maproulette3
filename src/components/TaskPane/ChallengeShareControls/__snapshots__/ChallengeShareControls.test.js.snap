// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not explode with null challenge 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChallengeShareControls />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders with props as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChallengeShareControls
    challenge={
      Object {
        "id": 123,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="share-control"
        >
          <CreatedButton
            quote="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>
        </div>,
        <div
          className="share-control"
        >
          <CreatedButton
            hashtags={
              Array [
                "maproulette",
              ]
            }
            title="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>
        </div>,
        <div
          className="share-control"
        >
          <CreatedButton
            body="http://127.0.0.1:3000/browse/challenges/123"
            onClick={[Function]}
            openWindow={false}
            subject="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>
        </div>,
      ],
      "className": "challenge-share-controls",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            quote="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>,
          "className": "share-control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>,
            "quote": "MapRoulette",
            "url": "http://127.0.0.1:3000/browse/challenges/123",
            "windowHeight": 400,
            "windowWidth": 550,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />,
              "className": "share-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "logoFillColor": "white",
                "round": true,
                "size": 26,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            hashtags={
              Array [
                "maproulette",
              ]
            }
            title="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>,
          "className": "share-control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>,
            "hashtags": Array [
              "maproulette",
            ],
            "title": "MapRoulette",
            "url": "http://127.0.0.1:3000/browse/challenges/123",
            "windowHeight": 400,
            "windowWidth": 550,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />,
              "className": "share-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "logoFillColor": "white",
                "round": true,
                "size": 26,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            body="http://127.0.0.1:3000/browse/challenges/123"
            onClick={[Function]}
            openWindow={false}
            subject="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>,
          "className": "share-control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "body": "http://127.0.0.1:3000/browse/challenges/123",
            "children": <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>,
            "onClick": [Function],
            "openWindow": false,
            "subject": "MapRoulette",
            "url": "http://127.0.0.1:3000/browse/challenges/123",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />,
              "className": "share-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "logoFillColor": "white",
                "round": true,
                "size": 26,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="share-control"
          >
            <CreatedButton
              quote="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>
          </div>,
          <div
            className="share-control"
          >
            <CreatedButton
              hashtags={
                Array [
                  "maproulette",
                ]
              }
              title="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>
          </div>,
          <div
            className="share-control"
          >
            <CreatedButton
              body="http://127.0.0.1:3000/browse/challenges/123"
              onClick={[Function]}
              openWindow={false}
              subject="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>
          </div>,
        ],
        "className": "challenge-share-controls",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              quote="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>,
            "className": "share-control",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>,
              "quote": "MapRoulette",
              "url": "http://127.0.0.1:3000/browse/challenges/123",
              "windowHeight": 400,
              "windowWidth": 550,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />,
                "className": "share-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "logoFillColor": "white",
                  "round": true,
                  "size": 26,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              hashtags={
                Array [
                  "maproulette",
                ]
              }
              title="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>,
            "className": "share-control",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>,
              "hashtags": Array [
                "maproulette",
              ],
              "title": "MapRoulette",
              "url": "http://127.0.0.1:3000/browse/challenges/123",
              "windowHeight": 400,
              "windowWidth": 550,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />,
                "className": "share-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "logoFillColor": "white",
                  "round": true,
                  "size": 26,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              body="http://127.0.0.1:3000/browse/challenges/123"
              onClick={[Function]}
              openWindow={false}
              subject="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>,
            "className": "share-control",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "body": "http://127.0.0.1:3000/browse/challenges/123",
              "children": <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>,
              "onClick": [Function],
              "openWindow": false,
              "subject": "MapRoulette",
              "url": "http://127.0.0.1:3000/browse/challenges/123",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />,
                "className": "share-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "logoFillColor": "white",
                  "round": true,
                  "size": 26,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders with props className in encapsulating div 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChallengeShareControls
    challenge={
      Object {
        "id": 123,
      }
    }
    className="my-test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="share-control"
        >
          <CreatedButton
            quote="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>
        </div>,
        <div
          className="share-control"
        >
          <CreatedButton
            hashtags={
              Array [
                "maproulette",
              ]
            }
            title="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>
        </div>,
        <div
          className="share-control"
        >
          <CreatedButton
            body="http://127.0.0.1:3000/browse/challenges/123"
            onClick={[Function]}
            openWindow={false}
            subject="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>
        </div>,
      ],
      "className": "challenge-share-controls my-test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            quote="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>,
          "className": "share-control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>,
            "quote": "MapRoulette",
            "url": "http://127.0.0.1:3000/browse/challenges/123",
            "windowHeight": 400,
            "windowWidth": 550,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />,
              "className": "share-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "logoFillColor": "white",
                "round": true,
                "size": 26,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            hashtags={
              Array [
                "maproulette",
              ]
            }
            title="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
            windowHeight={400}
            windowWidth={550}
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>,
          "className": "share-control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>,
            "hashtags": Array [
              "maproulette",
            ],
            "title": "MapRoulette",
            "url": "http://127.0.0.1:3000/browse/challenges/123",
            "windowHeight": 400,
            "windowWidth": 550,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />,
              "className": "share-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "logoFillColor": "white",
                "round": true,
                "size": 26,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CreatedButton
            body="http://127.0.0.1:3000/browse/challenges/123"
            onClick={[Function]}
            openWindow={false}
            subject="MapRoulette"
            url="http://127.0.0.1:3000/browse/challenges/123"
          >
            <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>
          </CreatedButton>,
          "className": "share-control",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "body": "http://127.0.0.1:3000/browse/challenges/123",
            "children": <span
              className="share-icon"
            >
              <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />
            </span>,
            "onClick": [Function],
            "openWindow": false,
            "subject": "MapRoulette",
            "url": "http://127.0.0.1:3000/browse/challenges/123",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                className=""
                logoFillColor="white"
                round={true}
                size={26}
              />,
              "className": "share-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "logoFillColor": "white",
                "round": true,
                "size": 26,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="share-control"
          >
            <CreatedButton
              quote="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>
          </div>,
          <div
            className="share-control"
          >
            <CreatedButton
              hashtags={
                Array [
                  "maproulette",
                ]
              }
              title="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>
          </div>,
          <div
            className="share-control"
          >
            <CreatedButton
              body="http://127.0.0.1:3000/browse/challenges/123"
              onClick={[Function]}
              openWindow={false}
              subject="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>
          </div>,
        ],
        "className": "challenge-share-controls my-test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              quote="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>,
            "className": "share-control",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>,
              "quote": "MapRoulette",
              "url": "http://127.0.0.1:3000/browse/challenges/123",
              "windowHeight": 400,
              "windowWidth": 550,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />,
                "className": "share-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "logoFillColor": "white",
                  "round": true,
                  "size": 26,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              hashtags={
                Array [
                  "maproulette",
                ]
              }
              title="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
              windowHeight={400}
              windowWidth={550}
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>,
            "className": "share-control",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>,
              "hashtags": Array [
                "maproulette",
              ],
              "title": "MapRoulette",
              "url": "http://127.0.0.1:3000/browse/challenges/123",
              "windowHeight": 400,
              "windowWidth": 550,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />,
                "className": "share-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "logoFillColor": "white",
                  "round": true,
                  "size": 26,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CreatedButton
              body="http://127.0.0.1:3000/browse/challenges/123"
              onClick={[Function]}
              openWindow={false}
              subject="MapRoulette"
              url="http://127.0.0.1:3000/browse/challenges/123"
            >
              <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>
            </CreatedButton>,
            "className": "share-control",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "body": "http://127.0.0.1:3000/browse/challenges/123",
              "children": <span
                className="share-icon"
              >
                <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />
              </span>,
              "onClick": [Function],
              "openWindow": false,
              "subject": "MapRoulette",
              "url": "http://127.0.0.1:3000/browse/challenges/123",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  className=""
                  logoFillColor="white"
                  round={true}
                  size={26}
                />,
                "className": "share-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "logoFillColor": "white",
                  "round": true,
                  "size": 26,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
